app/app.go

```diff
@@ -241,12 +252,15 @@ func (app *app) EndBlock(req tmtypes.RequestEndBlock) tmtypes.ResponseEndBlock {
 func (app *app) Commit() tmtypes.ResponseCommit {
 	app.trace("Commit")
 
-	err := app.cacheState.Write()
-	if err != nil {
-		app.log.Error("error when writing to cache")
+	if err := lease.ProcessLeases(app.cacheState); err != nil {
```

Comment:
why use cacheState here?

Reply:
thought was to keep all modifications in the ephemeral db until commit.  should we be using `commitState`?

Reply:
I thought the process lease routine sends transactions, but it modifies state directly. Using cache state is the correct thing to do 👍